core/src/test/java/org/kiwix/kiwixmobile/core/dao/NewBookmarksDaoTest.kt

```diff
@@ -174,9 +261,38 @@ internal class NewBookmarksDaoTest {
 
   @Test
   fun `deleteBookmarks removes specified bookmark item list`() {
-    val bookmarksList: List<BookmarkItem> = mockk(relaxed = true)
+    val id = 0L
+    val zimId = "zimId"
+    val zimName = "zimName"
+    val zimFilePath = "zimFilePath"
+    val bookmarkUrl = "bookmarkUrl"
+    val bookmarkTitle = "bookmarkTitle"
+    val favicon: String? = "favicon"
+    val bookmarkItem: BookmarkItem = mockk()
+    val bookmarksList: List<BookmarkItem> = listOf(bookmarkItem)
+    every { bookmarkItem.databaseId } returns id
+    every { bookmarkItem.zimId } returns zimId
+    every { bookmarkItem.zimName } returns zimName
+    every { bookmarkItem.zimFilePath } returns zimFilePath
+    every { bookmarkItem.bookmarkUrl } returns bookmarkUrl
+    every { bookmarkItem.bookmarkTitle } returns bookmarkTitle
+    every { bookmarkItem.favicon } returns favicon
     newBookmarksDao.deleteBookmarks(bookmarksList)
-    verify { box.remove(bookmarksList.map(::BookmarkEntity)) }
+    verify {
+      box.remove(
+        listOf(
+          bookmarkEntity(
+            id = id,
+            zimId = zimId,
+            zimName = zimName,
+            zimFilePath = zimFilePath,
+            bookmarkUrl = bookmarkUrl,
+            bookmarkTitle = bookmarkTitle,
+            favicon = favicon
+          )
+        )
+      )
+    }
```

Comment:
for the test below you need to expect data from
`bookDao.getFavIconAndZimFile(bookMark)`

don't mock data
```
val bookmarks: MutableList<Bookmark> = mockk(relaxed = true)
```

create it

Reply:
@akkywadhwa I was completely wrong here, forgot Bookmark is a TableModel ugh. mock is totally acceptable but you would want `listOf(mockk())`